xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererDX.cpp

```diff
@@ -213,17 +213,42 @@ COverlayImageDX::COverlayImageDX(CDVDOverlayImage* o)
     Load(rgba.data(), o->width, o->height, o->width * 4);
   }
 
-  if (o->source_width && o->source_height)
+  if (o->source_width > 0 && o->source_height > 0)
   {
-    float center_x = (0.5f * o->width + o->x) / o->source_width;
-    float center_y = (0.5f * o->height + o->y) / o->source_height;
-
-    m_align = ALIGN_VIDEO;
     m_pos = POSITION_RELATIVE;
-    m_x = center_x;
-    m_y = center_y;
-    m_width = static_cast<float>(o->width) / o->source_width;
-    m_height = static_cast<float>(o->height) / o->source_height;
+
+    m_x = (0.5f * o->width + o->x) / o->source_width;
+    m_y = (0.5f * o->height + o->y) / o->source_height;
+
+    if (o->source_height == static_cast<int>(rSource.Height()))
+    {
+      m_align = ALIGN_VIDEO;
+
+      m_width = static_cast<float>(o->width) / o->source_width;
+      m_height = static_cast<float>(o->height) / o->source_height;
+    }
+    else if (o->source_height > static_cast<int>(rSource.Height()))
+    {
+      // If the source_height (may referred to .idx VobSub file "size" param)
+      // is higher than the real video source height means that the
+      // video has been remuxed/cropped (removed black bars).
+      // Then we cannot align to video otherwise the subtitles will be deformed
+      // better align to screen by keeping the aspect-ratio
+      m_align = ALIGN_SCREEN_AR;
+
+      m_width = static_cast<float>(o->width);
+      m_height = static_cast<float>(o->height);
+      m_source_width = static_cast<float>(o->source_width);
+      m_source_height = static_cast<float>(o->source_height);
+    }
+    else
+    {
+      // Should never happens, in case stretch to screen resolution
```

Comment:
no reference to display resolution here, but subtitle res vs video (file) source res

i have now tested some mkv 4k (cropped and not cropped) and i have found PGS resolution is always set as 1080p
that is now rendered stretched, then i have to change again these conditions

use a kind of exclusion is more safe, but to try understand to avoid add it,
i need to know what values have these variables when the BD menu appear:
`o->source_height`
`rSource.Height`
can you provide these values?

for example with my sample BD i have:
`o->source_height` 1080
`rSource.Height` 1080

Reply:
Right now i can't test but i remember 3840, 2160 in `rSource` with Viuda Negra UHD BD. I'll confirm tomorrow.

Reply:
Now looks better